src/pool/srv_target.c

```diff
@@ -1028,6 +1028,58 @@ ds_pool_tgt_query_handler(crt_rpc_t *rpc)
 	crt_reply_send(rpc);
 }
 
+void
+ds_pool_tgt_fetch_hdls_handler(crt_rpc_t *rpc)
+{
+	struct pool_tgt_fetch_hdls_in	*in = crt_req_get(rpc);
+	struct pool_tgt_fetch_hdls_out	*out = crt_reply_get(rpc);
+	struct ds_pool			*pool;
+	int				 rc = DER_SUCCESS;
+	struct pool_iv_conn		*conn;
+
+	D_DEBUG(DF_DSMS, DF_UUID": processing rpc",
+		DP_UUID(in->tfi_pool_uuid));
+
+	pool = ds_pool_lookup(in->tfi_pool_uuid);
+	if (pool == NULL) {
+		D_ERROR("Failed to find pool "DF_UUID"\n",
+			DP_UUID(in->tfi_pool_uuid));
+		D_GOTO(out, rc = -DER_NONEXIST);
+	}
+
+	conn = in->tfi_hdls.iov_buf;
+
+	/* This while loop check first checks that there is room in the buffer
+	 * for the struct itself (not including the variable length creds field)
+	 * If there is room, it reaches into the buffer to get the size of the
+	 * creds field and repeats the check to ensure there is room
+	 */
+	while (((char *)conn - (char *)in->tfi_hdls.iov_buf +
+		sizeof(struct pool_iv_conn) <= in->tfi_hdls.iov_buf_len) &&
+	       ((char *)conn - (char *)in->tfi_hdls.iov_buf +
+		sizeof(struct pool_iv_conn) +
+		conn->pic_cred_size <= in->tfi_hdls.iov_buf_len)) {
```

Comment:
The loop should continue when "conn - iov_buf < iov_buf_len", and if above check fails, we should return error, right?

Reply:
Good idea - better to return an error than silently skip the last corrupted entry.